src/modules/esl-share/core/esl-share-action.ts

```diff
@@ -0,0 +1,29 @@
+import {memoize} from '../../esl-utils/decorators';
+import {ESLShareActionRegistry} from './esl-share-action-registry';
+
+import type {ESLShareButton} from './esl-share-button';
+
+export type ActionType = (new($button: ESLShareButton) => ESLShareBaseAction) & typeof ESLShareBaseAction;
+
+export abstract class ESLShareBaseAction {
+  public static readonly is: string;
+
+  @memoize()
```

Comment:
Why do we memoizing literal ?

Reply:
Some of the base action's descendants may not be available for the client. I find it easier to override the getter than the constructor.
Or do you have any suggestions for this case?

Reply:
Sorry, misunderstood. Fixed